### PR TITLE
[Update 10] Add GraphQL Changes to the Release Note

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose

$Subject

### CODE REVIEW

File: downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md [hunk 1]
```diff
@@ -208,3 +208,35 @@ public function main() {
         string|crypto:PublicKey certFile?;
     |};
     ```
+
+#### `graphql` package
```
Comment (MaryamZi) [inline downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md:212]: This is not a backward-incompatible change, right? Currently listed under that section.

File: downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md [hunk 2]
```diff
@@ -208,3 +208,35 @@ public function main() {
...
+
+- Add support to GraphQL query complexity analysis, which can be used to enhance the GraphQL service security by mitigating the risk of denial-of-service attacks. With this update, the `graphql:ServiceConfig` record now includes a new field `queryComplexityConfig` to configure the query complexity analysis.
```
Comment (MaryamZi) [inline downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md:214]:
```suggestion
- Added support for GraphQL query complexity analysis, which can be used to enhance the GraphQL service security by mitigating the risk of denial-of-service attacks. With this update, the `graphql:ServiceConfig` record now includes a new field named `queryComplexityConfig` to configure query complexity analysis.
```

File: downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md [hunk 3]
```diff
@@ -208,3 +208,35 @@ public function main() {
...
+        // Following are the default values, which can be overridden.
+        queryComplexityConfig: {
+            maxComplexity: 100,
+            defaultFieldComplexity: 1,
+            warnOnly: false
+        }
+    }
+    service graphql:Service /graphql on new graphql:Listener(9090) {
+
+        // Defines a custom complexity value for the field
```
Comment (MaryamZi) [inline downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md:229]:
```suggestion
        // Defines a custom complexity value for the field.
```

File: downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md [hunk 4]
```diff
@@ -208,3 +208,35 @@ public function main() {
         string|crypto:PublicKey certFile?;
     |};
     ```
+
+#### `graphql` package
+
+- Add support to GraphQL query complexity analysis, which can be used to enhance the GraphQL service security by mitigating the risk of denial-of-service attacks. With this update, the `graphql:ServiceConfig` record now includes a new field `queryComplexityConfig` to configure the query complexity analysis.
+
+    ```ballerina
```
Comment (MaryamZi) [inline downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md:216]: Usually for library modules we seem to be adding just a one or two line description, without including examples. This is a bit of a deviation. https://github.com/ballerina-platform/ballerina-dev-website/blob/release-2201.10.0/downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md#new-features-2

I am +1 for adding examples though.

File: downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md (hunk 1 above)
Comment (ThisaruGuruge) [reply downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md]: Mistake. Fixed with 544bef54d20593b423108a1d5987f238f98ac864

File: downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md (hunk 2 above)
Comment (ThisaruGuruge) [reply downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md]: Fixed with 544bef54d20593b423108a1d5987f238f98ac864

File: downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md (hunk 4 above)
Comment (ThisaruGuruge) [reply downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md]: Removed the example to keep the consistency. We have to have a discussion on whether to add the examples here or not.

File: downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md [hunk 5]
```diff
@@ -88,6 +88,10 @@ To view bug fixes, see the [GitHub milestone for Swan Lake Update 10 (2201.10.0)
 - Introduced support for parsing XML with record types with default values as the expected type, using the default values where required (i.e., if a value corresponding to the record field is not present in the XML value).
 - Introduced the option to choose between semantic and syntactic equality of XML elements and attributes.
 
+#### `graphql` package
+
+- Added support for GraphQL query complexity analysis, which can be used to enhance the GraphQL service security by mitigating the risk of denial-of-service attacks. With this update, the `graphql:ServiceConfig` record now includes a new field named `queryComplexityConfig` to configure the query complexity analysis.
```
Comment (MaryamZi) [inline downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md:93]:
```suggestion
- Added support for GraphQL query complexity analysis, which can be used to enhance GraphQL service security by mitigating the risk of denial-of-service attacks. With this update, the `graphql:ServiceConfig` annotation now includes a new field named `queryComplexityConfig` to configure query complexity analysis.
```

File: downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md [hunk 6]
```diff
@@ -241,7 +245,6 @@ public function main() returns error? {
 #### `xslt` package
 
 - Added parameter passing support for XSLT transformations.
```
Comment (ThisaruGuruge) [inline downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md:247]:
```suggestion
- Added parameter passing support for XSLT transformations.

```

File: downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md [hunk 7]
```diff
@@ -119,6 +119,10 @@ To view bug fixes, see the [GitHub milestone for Swan Lake Update 10 (2201.10.0)
 - Introduced support for parsing XML with record types with default values as the expected type, using the default values where required (i.e., if a value corresponding to the record field is not present in the XML value).
 - Introduced the option to choose between semantic and syntactic equality of XML elements and attributes.
 
+#### `graphql` package
```
Comment (MaryamZi) [inline downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md:122]: Shall we move this to after the data.yaml section to maintain alphabetic order?

Comment (ThisaruGuruge) [reply downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md]: Updated.

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**